client/db/bolt/db.go

```diff
@@ -1629,6 +1650,79 @@ func loadMatchBucket(mBkt *bbolt.Bucket, excludeCancels bool) (*dexdb.MetaMatch,
 	}, nil
 }
 
+// errFound is used to break ForEach early when a match is found.
+var errFound = errors.New("found")
+
+// hasTradeMatch checks if an order has any non-cancel trade matches.
+// A trade match has a non-empty Address (not a cancel match for maker)
+// and has InitSig (not a cancel match for taker).
+// Returns true immediately upon finding the first trade match (early exit).
+// Checks archived bucket first since canceled/revoked orders typically have
+// settled matches there. Also checks active bucket for matches still settling.
+func (db *BoltDB) hasTradeMatch(oid order.OrderID) bool {
+	found := false
+	err := db.matchesView(func(mb, archivedMB *bbolt.Bucket) error {
+		// Check archived bucket first (most common case for canceled orders)
+		for _, bucket := range []*bbolt.Bucket{archivedMB, mb} {
+			if bucket == nil {
+				continue
+			}
+			isArchived := bucket == archivedMB
+			err := bucket.ForEach(func(k, _ []byte) error {
+				mBkt := bucket.Bucket(k)
+				if mBkt == nil {
+					return nil
+				}
+				// Check if match belongs to this order
+				if !bytes.Equal(mBkt.Get(orderIDKey), oid[:]) {
+					return nil
+				}
+
+				// Decode match (lightweight - only UserMatch)
+				matchB := mBkt.Get(matchKey)
+				if matchB == nil {
+					return nil
+				}
+				match, _, err := order.DecodeMatch(matchB)
+				if err != nil {
+					db.log.Errorf("DecodeMatch error for match %x: %v", k, err)
+					return nil // Continue checking other matches
+				}
+
+				// Cancel match for maker has empty Address
+				if match.Address == "" {
+					return nil
+				}
+
+				// Cancel match for taker has no InitSig and status MatchComplete
+				// Only check this for archived bucket since inactive matches
+				// (MatchComplete without InitSig) should be archived
+				if isArchived && match.Status == order.MatchComplete {
+					proofB := mBkt.Get(proofKey)
+					if len(proofB) > 0 {
+						proof, _, err := dexdb.DecodeMatchProof(proofB)
+						if err == nil && len(proof.Auth.InitSig) == 0 {
+							return nil // Cancel match for taker
+						}
+					}
+				}
```

Comment:
I think checking the address only is enough.

Reply:
I'm pretty sure before the match is settled, the address is cleared somewhere, will check again.